### PR TITLE
Load EasyMDE/CodeMirror dynamically, remove RequireEasyMDE

### PR DESCRIPTION
**Pull request by wxiaoguang**
As discussed in:

* https://github.com/go-gitea/gitea/pull/17926

This PR makes frontend load EasyMDE/CodeMirror dynamically, and removes `RequireEasyMDE`.

### CODE REVIEW

**Comment by silverwind** [comment]
I think we can also remove the static CodeMirror copy in `public/vendor/plugins/codemirror`, which also means EasyMDE will now use the newer CodeMirror version from the npm package.

**Comment by wxiaoguang** [comment]
> I think we can also remove the static CodeMirror copy in `public/vendor/plugins/codemirror`, which also means EasyMDE will now use the newer CodeMirror version from the npm package.

Maybe we can try that in future, but for now, it's quite tricky, some CodeMirror plugins have to be loaded from that directory directly .... I have left comment there, and there are two issue links for CodeMirror.

https://github.com/go-gitea/gitea/pull/18069/files#diff-5eca01f7999043471c646505421dd1362dc17aeba4f600b72ac1c1e76864d7e1R31

**Comment by zeripath** [comment]
> I think we can also remove the static CodeMirror copy in `public/vendor/plugins/codemirror`, which also means EasyMDE will now use the newer CodeMirror version from the npm package.

I'm fairly sure I tried this when I migrated to EasyMDE in the first place but was unable to make this work. I probably added a comment in #13333 somewhere about this.

**Comment by silverwind** [comment]
Right, I remember CodeMirrors plugins being generally incompatible with Webpack's loading mechanism, so that must be the reason for the vendored copy.

File: web_src/js/features/repo-issue.js
Comment (silverwind) [inline web_src/js/features/repo-issue.js:443]: IIFE needed for this `preventDefault`? I think but I think the code may as well work with removal of this line it as button click should not have a default action.

Comment (silverwind) [reply web_src/js/features/repo-issue.js:443]: Also [this rule](https://github.com/github/eslint-plugin-github/blob/main/docs/rules/async-preventdefault.md) should warn on such issues but I guess it can only work with standard `addEventListener`, not jQuery. We really need to get rid of jQuery event listeners :)

Comment (wxiaoguang) [reply web_src/js/features/repo-issue.js:443]: IIRC, it would submit a form?

**Comment by wxiaoguang** [comment]
> Edit: [codemirror/CodeMirror#4838 (comment)](https://github.com/codemirror/CodeMirror/issues/4838#issuecomment-313134082) is related. So I think we'd need to manually load the markdown mode (the only mode we use it for) and other plugins in webpack, similar to [codemirror/CodeMirror#5484 (comment)](https://github.com/codemirror/CodeMirror/issues/5484#issue-338185331).

And this time I added these comments into code. Next time a new developer won't be surprised. 😋

**Comment by wxiaoguang** [comment]
How do you think about this PR? Can we move on?

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
@go-gitea/maintainers  I will resolve conflicts after #18161 is merged. 

But I still need one LGTM, hopefully we can move on, soon.

File: web_src/js/features/comp/EasyMDE.js
Comment (Gusted) [inline web_src/js/features/comp/EasyMDE.js]: Any particular reason for this particular order? I think the browser would appreciate if the url was known when adding it to the DOM.

File: web_src/js/features/comp/EasyMDE.js
Comment (Gusted) [inline web_src/js/features/comp/EasyMDE.js:43]: Gitea's JS don't have a "general" error handler, e.g. in the sense that when a error happens that the user has no clue that it happend. So should we go fancy on using Errors?

File: web_src/js/features/comp/EasyMDE.js
Comment (Gusted) [inline web_src/js/features/comp/EasyMDE.js:69]: I'm pretty sure their is a eslint-rule for this kind of code-style. Maybe it should be enabled.

File: web_src/js/features/comp/EasyMDE.js
Comment (wxiaoguang) [reply web_src/js/features/comp/EasyMDE.js]: Oh yeah, the order doesn't matter, just a personal preference 😂

File: web_src/js/features/comp/EasyMDE.js
Comment (wxiaoguang) [reply web_src/js/features/comp/EasyMDE.js:43]: Before: no failure check.
Now: we throw an error.

So it's still a little better than before. 

I wouldn't expect to expose these errors to end users, but prefer to use the checks to help developers to find potential bugs ASAP in future.

**Comment by lunny** [comment]
L-G-T-M bot countered the author's :)

**Review by Gusted — APPROVED** [review]
You got mine approval, I think it's also nice if you can get one from silverwind.

**Comment by wxiaoguang** [comment]
Merged with main branch and conflicts are resolved. If there is no objection in 2 days, we could get this this one merged and move on.

File: web_src/js/features/comp/EasyMDE.js
Comment (wxiaoguang) [reply web_src/js/features/comp/EasyMDE.js]: Updated.

**Review by Gusted — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Pull request merged**
